Sources/WWDCNotes/WWDCNotes.docc/WWDC24/WWDC24-10166-Build-compelling-spatial-photo-and-video-experiences.md

```diff
@@ -8,11 +8,113 @@ Learn how to adopt spatial photos and videos in your apps. Explore the different
+
+### Detecting spatial media
+
+Detection Methods
+- PhotosPicker. Can filter library to show only `.spatialMedia` assets.
+- PhotoKit. Pass `PHAssetMediaSubtype.spatialMedia` to `fetchAssets` call. Can do just spatial photos or spatial videos.
+- AVAssetPlaybackAssistant. `.playbackConfigurationOptions` can be configured for spatial assets.
+
+Presentation Options
+- PreviewApplication API (spawn Quicklook scene). See "What's new in Quick Look for spatial computing".
+- Element Fullscreen API (Javascript). Open spatial photos in Safari. See "Optimize for the Spatial Web".
+- AVPlayerViewController. Supports 2D and 3D video content, also HTTP Live Streaming for spatial video. Only displays as 3D if full screen. Shows 3D video, not spatial video. Use PreviewApplication if you need spatial video display.
+
+## Create custom spatial media
+
+[File format discussion](https://developer.apple.com/videos/play/wwdc2024/10166/?time=790).
+Two sample projects provided, for spatial video and spatial photos.
+
+[Spatial metadata:](https://developer.apple.com/videos/play/wwdc2024/10166/?time=821)
+- Projection. Spatial photos and videos always use the rectilinear projection.
+- Baseline, Field of View: camera properties.
+- Disparity adjustment
+
+Example: stereoscopic image of a hummingbird.
+
+Note: the following metadata discussion is put into practice in the [Writing spatial photos](https://developer.apple.com/documentation/ImageIO/writing-spatial-photos) sample app.
+
+For optimal image characteristics the left and right images should be stereo rectified, have optical axis alignment, and have no vertical disparity.
+Rectilinear: straight lines in world are straight lines on image.
+Horizontal baseline: 64 mm similar to human eye, 32 mm good for closeups, larger than 64 mm good for stereo landscape photography. 
+
+FOV greater than 90 degrees is inefficient for rectilinear capture, 60 degrees used for this analysis. 
+
+Images must be coplanar (or stereorectified in post-production).
+
+Cropping: center axis of cameras should be at centers of images to keep good alignment. Vertical disparity, if present, makes images uncomfortable to view (same feature should have same Y coordinate in each eye).
+
+Vision Pro renderer uses the metadata to construct a camera model, renders different image for each eye.
+
+Horizontal disparity adjustment: a hint to the renderer on where to place the zero parallax plane. Sometimes called convergence adjustment. Move left/right frames horizontally to adjust disparity. This controls perception of how far away the 3D scene appears to be.
+
+## Wrap-up. Covered:
+- multiple types of stereo media
+- capture, detect, and display spatial media
+- create your own spatial media
+
+
+
+
+
+
+
+
```

Comment:
Why are there so many new lines? Any specific reason? It's certainly not needed.

Reply:
Done